src/mmrm.cpp

```diff
@@ -52,30 +53,39 @@ Type objective_function<Type>::operator() ()
   vector<Type> y_vec_tilde = vector<Type>::Zero(y_vector.rows());
   // Sum of the log determinant will be incrementally calculated here.
   Type sum_log_det = 0.0;
-
   // Create the lower triangular Cholesky factor of the visit x visit covariance matrix.
-  matrix<Type> covariance_lower_chol = get_covariance_lower_chol<Type>(theta, n_visits, cov_type);
-
+  matrix<Type> covariance_lower_chol = matrix<Type>::Zero(n_visits, n_visits * n_groups);
+  int covariance_size = theta.size() / n_groups;
+  for (int i = 0; i < n_groups; i++) {
+    matrix<Type> lower_chol = get_covariance_lower_chol<Type>(theta.segment(i * n_groups, covariance_size), n_visits, cov_type);
+    for (int j = 0; j < n_visits; j++) {
+      covariance_lower_chol.col(j + i * n_visits) = lower_chol.col(j);
+    }
+  }
+  
   // Go through all subjects and calculate quantities initialized above.
   for (int i = 0; i < n_subjects; i++) {
     // Start index and number of visits for this subject.
     int start_i = subject_zero_inds(i);
     int n_visits_i = subject_n_visits(i);
-
     // Obtain Cholesky factor Li.
     matrix<Type> Li;
+    matrix<Type> lower_chol = matrix<Type>::Zero(n_visits, n_visits);
+    for (int j = 0; j < n_visits; j++) {
+      lower_chol.col(j) = covariance_lower_chol.col(j + subject_groups(i) * n_visits);
```

Comment:
I think this should be able to do in a simpler way using the matrix subsetting in TMB, please check in https://kaskr.github.io/adcomp/matrix_arrays_8cpp-example.html